[CSL-1005] Purify cardano-sl from SysStart leftovers

ChShersh opened the pull request:
This PR remove leftovers after time-lord deletion. Some TODO's though are still left.

gromakovsky commented:
@ChShersh merge master into this branch, it should fix CI failures.

gromakovsky commented:
(or rebase, yes)

gromakovsky commented on src/Pos/CLI.hs:
```diff
@@ -124,7 +125,7 @@ data CommonArgs = CommonArgs
     , richPoorDistr      :: !(Maybe (Int, Int, Integer, Double))
     , expDistr           :: !Bool
-    , sysStart           :: !Timestamp
+    , sysStart           :: !Timestamp  -- TODO: remove it?
```
No, why? System start is usually passed via CLI nowadays.

gromakovsky commented on src/Pos/Launcher/Runner.hs:
```diff
@@ -173,6 +173,7 @@ runRawRealMode res np@NodeParams {..} sscnp listeners outSpecs (ActionSpec actio
   where
     LoggingParams {..} = bpLoggingParams npBaseParams
 
+-- TODO: remove this function?
```
No, why?

ChShersh replied:
Because it uses `sysStart`. It's not clear for me what is _leftover_ so I'm asking about every `sysStart`

gromakovsky replied:
System start time is still necessary. Formerly in development mode nodes obtained it by having one time-lord and multiple time-slaves. So we have some workers, listeners, messages. Now we want it to be passed via CLI and don't want to have old machinery.

gromakovsky commented on src/Pos/Launcher/Runner.hs:
```diff
@@ -344,14 +345,14 @@ setupLoggers params = setupLogging =<< getRealLoggerConfig params
     if Const.isDevelopment
-    then mergeLs [ stubListenerOneMsg (Proxy :: Proxy SysStartResponse)
-                 , sysStartReqListener sysStart] `mappendPair` ls
+    then mergeLs [ stubListenerOneMsg (Proxy :: Proxy MsgGetHeaders)] `mappendPair` ls
```
I don't understand this place. Can someone more familiar with this code comment?
@georgeee @Martoon-00 @volhovM

volhovM replied:
The listener that responds with system start time is removed. What's the question?

gromakovsky replied:
1. There is
>  +-- TODO: remove this function and all it's calls? 
Do you know the answer?
2. How did `MsgGetHeaders` appear here? It's very counterintuitive.

ChShersh replied:
@gromakovsky @volhovM I removed `SysStartResponse` data type. So I can use it here. I don't want to remove this function yet because removal of it will also remove a lot of functions. I need to put something that is `Message` after `Proxy` for this code to compile. Obviously, this should be removed. I just asking: is there any sense in old function. Should I remove it as whole or not?

gromakovsky replied:
@ChShersh I am 99% sure you can remove this `addDevListeners` stuff and make code cleaner.

gromakovsky commented on src/smart-generator/Main.hs:
```diff
@@ -252,7 +252,7 @@ main = do
     bracketResources baseParams $ \res -> do
 
-        let systemStart = CLI.sysStart goCommonArgs
+        let systemStart = CLI.sysStart goCommonArgs  -- TODO: remove it?
```
No.

ChShersh commented:
Okay, I rebased on `master`, all leftovers are now currently removed. Also I've added some minor change. Just noticed this function while I was refactoring. Decided to fix it as well.

gromakovsky commented on infra/Pos/Communication/Types/Protocol.hs:
```diff
@@ -192,7 +191,7 @@ instance Monoid OutSpecs where
 
 mergeLs :: [(ListenerSpec m, OutSpecs)] -> ([ListenerSpec m], OutSpecs)
-mergeLs = second mconcat . (map fst &&& map snd)
+mergeLs = second mconcat . unzip
```
kek

gromakovsky commented:
@ChShersh did you test it locally? Do it please (if you didn't). At least that 3 nodes launched with `scripts/launch.sh` generate blocks and have consensus about slotting.

ChShersh commented:
@gromakovsky Confirming. It works.

gromakovsky commented:
@ChShersh conflicts appeared, rebase/merge pls.

gromakovsky commented:
@ChShersh ping.

ChShersh commented:
@gromakovsky Rebased. So we're waiting for build. I hope there won't more conflicts with `master` :)
